### PR TITLE
Added more generic array examples

### PR DESCRIPTION
**Pull request by ansharlubis**
The examples where size of the arrays are variables (T[n]) do not compile into LLVM.

### CODE REVIEW

**Comment by ansharlubis** [comment]
@certik I added a couple more generic array examples based on #942

**Comment by certik** [comment]
Let's enable CPython tests. Also, there are error messages from LPython, so it means we need to improve LPython to compile these?

**Comment by ansharlubis** [comment]
There does still seem to be some issues with the generated functions. I'll try to look into it.

**Comment by ansharlubis** [comment]
@certik I got the additional array examples to work in both CPython and LLVM. I removed the example that uses size(..) in the type annotation.

File: src/libasr/pass/instantiate_template.cpp [hunk 1]
```diff
@@ -26,16 +26,16 @@ class FunctionInstantiator : public ASR::BaseExprStmtDuplicator<FunctionInstanti
         new_func_name{new_func_name}
         {}
 
-    ASR::asr_t* instantiate_Function(ASR::Function_t &x) {
+    ASR::asr_t* instantiate_Function(ASR::Function_t *x) {
```
Comment (certik) [inline src/libasr/pass/instantiate_template.cpp:29]: What is the motivation of this change? Can `x` be `nullptr`?

**Review by certik — APPROVED** [review]
I think this is good enough to merge, I left a comment above.

Thanks!

File: src/libasr/pass/instantiate_template.cpp (hunk 1 above)
Comment (ansharlubis) [reply src/libasr/pass/instantiate_template.cpp:29]: I think I made some changes that requires `&x` but then I made more changes that actually work with `*x`. I don't really have any reason for it. Should I revert it back?
